project1-sps/script.js

```diff
@@ -1,4 +1,53 @@
-var main = function (input) {
-  var myOutputValue = 'hello world';
+//Default output
+var myOutputValue = `Hello World`;
+//get auto generated Computer move
+var main = function (player) {
+  // Generate a decimal from 0 through 3, inclusive of 0 and exclusive of 3.
+  var randomNum = Math.random() * 3;
+  // Remove the decimal with the floor operation.
+  // This will be an integer from 0 to 3 inclusive.
+  var randomRound = Math.floor(randomNum);
+  // Add 1 to get valid result of 1 through 3 inclusive.
+  var comMove = randomRound + 1;
+  if (comMove == 1) {
+    comMove = `Scissor`;
+  } else if (comMove == 2) {
+    comMove = `Paper`;
+  } else {
+    comMove = `Stone`;
+  }
+
+  //Logic on Cases
+  if (comMove === player) {
+    myOutputValue = `You choose ${player} ! AI choose ${comMove} ! It is a draw! `;
+  } else if (
+    (player === `Scissor` && comMove === `Paper`) ||
+    (player === `Paper` && comMove === `Stone`) ||
+    (player === `Stone` && comMove === `Scissor`)
+  ) {
+    myOutputValue = `You choose ${player} ! AI choose ${comMove} ! You Win! `;
+  } else {
+    myOutputValue = `You choose ${player} ! AI choose ${comMove} ! You Lose! `;
+  }
   return myOutputValue;
 };
```

Comment:
Concise game logic. I like it. It tests your understanding of if, else if and else